db/migrate/20170301150018_delete_orphaned_data.rb

```diff
@@ -0,0 +1,48 @@
+class DeleteOrphanedData < ActiveRecord::Migration
+  def change
+    #delete works for which there is no collection
+    Work.all.each {|w| w.destroy unless w.collection}
+    
+    #delete pages for which there are no works
+    Page.all.each {|p| p.destroy unless p.work}
+    
+    #delete document sets for which there is no collection
+    DocumentSet.all.each {|d| d.destroy unless d.collection}
+
+    #delete notes for which there is no collection
+    Note.all.each {|n| n.destroy unless n.collection}
```

Comment:
Thinking more.  People delete works, pages, and collections and any of the above could leave behind orphaned notes.  I think this needs to be fleshed out more to get the logic right.  Maybe the order is important too -- start with the highest things in the hierarchy (collections => works & document sets => pages => notes) so that the "parent" of a particular thing is gone by the time we check for it?